addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java

```diff
@@ -198,68 +225,123 @@ public void setStatusInfo(ThingStatus status, ThingStatusDetail statusDetail, St
         super.updateStatus(status, statusDetail, description);
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public FritzahaWebInterface getWebInterface() {
         return this.connection;
     }
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
     public void addDeviceList(DeviceModel device) {
         try {
             logger.debug("set device model: {}", device);
             ThingUID thingUID = getThingUID(device);
-            if (getThing() != null) {
+            Thing thing = getThing();
+            if (thing != null) {
                 logger.debug("update thing {} with device model: {}", thingUID, device);
-                if (device.isTempSensor() && device.getTemperature() != null) {
-                    updateTemperatureChannel(device.getTemperature().getCelsius());
-                }
-                if (device.isPowermeter() && device.getPowermeter() != null) {
-                    updateEnergyChannel(device.getPowermeter().getEnergy());
-                    updatePowerChannel(device.getPowermeter().getPower());
+                updateThingFromDevice(thing, device);
+            }
+        } catch (Exception e) {
+            logger.error("{}", e.getLocalizedMessage(), e);
+        }
+    }
+
+    /**
+     * Updates things from device model.
+     *
+     * @param thing Thing to be updated.
+     * @param device Device model with new data.
+     */
+    private void updateThingFromDevice(Thing thing, DeviceModel device) {
+        if (thing == null || device == null) {
+            throw new IllegalArgumentException("thing or device is null, cannot perform update");
+        }
+        if (device.getPresent() == 1) {
+            thing.setStatusInfo(new ThingStatusInfo(ThingStatus.ONLINE, ThingStatusDetail.NONE, null));
+            if (device.isTempSensor() && device.getTemperature() != null) {
+                updateThingChannelState(thing, CHANNEL_TEMP, new DecimalType(device.getTemperature().getCelsius()));
+            }
+            if (device.isPowermeter() && device.getPowermeter() != null) {
+                updateThingChannelState(thing, CHANNEL_ENERGY, new DecimalType(device.getPowermeter().getEnergy()));
+                updateThingChannelState(thing, CHANNEL_POWER, new DecimalType(device.getPowermeter().getPower()));
+            }
+            if (device.isSwitchableOutlet() && device.getSwitch() != null) {
+                updateThingChannelState(thing, CHANNEL_MODE, new StringType(device.getSwitch().getMode()));
```

Comment:
No we can't. This is based on the type of the FRITZ! devices. There is the FRITZ!Box which will be a `Bridge`. Then there are several devices like FRITZ! DECT 200 or Comet DECT which will be `Things`. And there is a third component FRITZ!Powerline - similiar to a WLAN repeater but also usable as standalone unit - which is a mixture of a `Bridge` and a `Thing` (see [HandlerFactory](https://github.com/openhab/openhab2-addons/blob/master/addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/internal/HandlerFactory.java#L60)) - and only this type of device will call the `updateThingFromDevice` method because it may have its own polling job.

_MEMO to myself: Maybe we should introduce a third handler for this device._